src/init/mod.rs

```diff
@@ -143,32 +189,31 @@ fi"#,
 /* This function (called when `--print-full-init` is passed to `starship init`)
 prints out the main initialization script */
 pub fn init_main(shell_name: &str) -> io::Result<()> {
-    let starship_path = path_to_starship()?.replace("\"", "\"'\"'\"");
-
-    let setup_script = match shell_name {
-        "bash" => Some(BASH_INIT),
-        "zsh" => Some(ZSH_INIT),
-        "fish" => Some(FISH_INIT),
-        "powershell" => Some(PWSH_INIT),
-        "ion" => Some(ION_INIT),
+    let starship_path = StarshipPath::init()?;
+
+    match shell_name {
+        "bash" => print_script(BASH_INIT, &starship_path.sprint_unix()?),
+        "zsh" => print_script(ZSH_INIT, &starship_path.sprint_unix()?),
+        "fish" => print_script(FISH_INIT, &starship_path.sprint_unix()?),
+        "powershell" => print_script(PWSH_INIT, &starship_path.sprint()?),
+        "ion" => print_script(ION_INIT, &starship_path.sprint()?),
```

Comment:
I think `powershell` and `ion` should both use `sprint_unix`. PowerShell has a linux version and ion doesn't even run on windows.

Reply:
No. `sprint` directly return the platform native path string, which is the normal case.

`sprint_unix` return converted(on Windows)/native(on other platforms)  path string, is only used for some shells(bash/zsh/fish/etc) that runs on Windows through posix compatibility layer provided by cygwin.To be more precise, I renamed it to `sprint_posix`.

These shells can only accept posix path string (`/c/Windows/`) not native (`C:\Windows`), which cause the issues.

PowerShell is natively cross-platform and can handle native path on every platform. Ion though have not yet supported Windows, we shouldn't do additional thing except they decide to support Windows via cygwin runtime.

By the way, bash have been additionally patched to handle both Windows and posix path string, that's why people used git-bash with starship without problem, but zsh/fish not. To be consistent, we don't rely on this behavior and also apply conversion to it.

Reply:
Wouldn't PowerShell running via the Cygwin compatibly layer on windows still need the posix-style path? Also ion might not officially support windows, but I expect running it with Cygwin is possible.

Reply:
Cygwin is a compatibility layer specifically designed for C programs written against the posix api.
Powershell is built on .NET platform, which is intrinsic cross platform. I don't know if there're any way or any reason to build .NET runtime against cygwin...
Rust [did not support cygwin](https://users.rust-lang.org/t/does-rust-work-in-cygwin-if-so-how-can-i-get-it-working/25735) as a target, nor need to. Porting Ion to Windows is more possible than building it against cygwin since rust is a modern programming language.
Anyhow, since Ion currently unable to run on Windows, whether patch it or not makes no sense. I think keep it intact is more intuitive and avoid unexpected behavior in the future...